staging/src/k8s.io/apimachinery/pkg/util/proxy/transport.go

```diff
@@ -193,7 +191,11 @@ func rewriteHTML(reader io.Reader, writer io.Writer, urlRewriter func(string) st
 			if urlAttrs, ok := atomsToAttrs[token.DataAtom]; ok {
 				for i, attr := range token.Attr {
 					if urlAttrs.Has(attr.Key) {
-						token.Attr[i].Val = urlRewriter(attr.Val)
+						url, err := url.Parse(attr.Val)
+						if err != nil {
+							return err
```

Comment:
this means any html content returned through the proxy containing an attribute with a URL that can't be parsed now errors? that does not match my expectations for the proxy behavior

Reply:
yes.

If parsing fails, I think we only have two choices, either we error or we skip the attribute since rewriting an unescaped URL will cause the same problems as what we had before.

That said, would skipping the attribute be an expected behavior to you? My only fear is that we may end up hiding errors that would give valuable information to the clients in case something is incorrect with the `Location` header they set.

Reply:
we should definitely not turn an inability to understand content in the returned body into a fatal error

Reply:
make sense. I updated the code to now skip rewriting the URL if isn't valid.